ntoskrnl/ex/harderr.c

```diff
@@ -629,37 +660,17 @@ NtRaiseHardError(IN NTSTATUS ErrorStatus,
         _SEH2_END;
 
         /* Call the system function directly, because we probed */
-        ExpRaiseHardError(ErrorStatus,
-                          NumberOfParameters,
-                          UnicodeStringParameterMask,
-                          SafeParams,
-                          ValidResponseOptions,
-                          &SafeResponse);
-    }
-    else
-    {
-        /* Reuse variable */
-        SafeParams = Parameters;
-
-        /*
-         * Call the Executive Function. It will probe and copy pointers to
-         * user-mode
-         */
-        ExRaiseHardError(ErrorStatus,
-                         NumberOfParameters,
-                         UnicodeStringParameterMask,
-                         SafeParams,
-                         ValidResponseOptions,
-                         &SafeResponse);
-    }
-
-    /* Check if we were called in user-mode */
-    if (PreviousMode != KernelMode)
-    {
-        /* That means we have a buffer to free */
+        Status = ExpRaiseHardError(ErrorStatus,
```

Comment:
Not really your doing, but saving the Status here makes it more obvious: if this function fails, SafeResponse may be uninitialized down in L666/677